pkg/action/pull.go

```diff
@@ -138,6 +139,37 @@ func (p *Pull) Run(chartRef string) (string, error) {
 		return out.String(), err
 	}
 
+	// Print a pull summary for repo mode and non-OCI downloads.
+	// Some repos (e.g., Bitnami) resolve --repo pulls to oci:// refs, but that
+	// path does not go through the registry client's summary printer, so we emit
+	// a consistent summary here. We mirror the OCI output format:
+	//
+	//   Pulled: <chart>:<version>
+	//   Digest: sha256:<digest>
+	//
+	// For HTTP/repo pulls, the digest is the SHA-256 of the downloaded .tgz archive.
+	// For direct OCI pulls, the registry client already prints "Pulled:" and
+	// "Digest:" (manifest digest), so we do not print here to avoid duplicates.
+	if p.RepoURL != "" || !registry.IsOCI(downloadSourceRef) {
```

Comment:
Not a big fan of this since it feels a bit hacky, but with OCI auto-resolving like Bitnami I was short on alternatives.

Reply:
Thanks @sabre1041 you are right. It is misleading. Because we use `sha256.New()` it is a [checksum](https://pkg.go.dev/crypto/sha256#New). 

Changed in https://github.com/helm/helm/pull/31206/commits/83789b112b43828e17bfbf6a9a1c964278a54f77